rcl/src/rcl/publisher.c

```diff
@@ -304,6 +306,40 @@ rcl_publisher_assert_liveliness(const rcl_publisher_t * publisher)
   return RCL_RET_OK;
 }
 
+rcl_ret_t
+rcl_publisher_wait_for_all_acked(const rcl_publisher_t * publisher, rcl_duration_value_t timeout)
+{
+  if (!rcl_publisher_is_valid(publisher)) {
+    return RCL_RET_PUBLISHER_INVALID;  // error already set
+  }
+
+  rmw_time_t rmw_timeout;
+  if (timeout > 0) {
+    rmw_timeout.sec = RCL_NS_TO_S(timeout);
+    rmw_timeout.nsec = timeout % 1000000000;
+  } else if (timeout < 0) {
+    rmw_time_t infinite = RMW_DURATION_INFINITE;
+    rmw_timeout = infinite;
```

Comment:
```suggestion
    rmw_timeout = RMW_DURATION_INFINITE;
```

Reply:
```c++
#define RMW_DURATION_INFINITE {9223372036LL, 854775807LL}
```
So `RMW_DURATION_INFINITE` cannot be assigned to rmw_timeout.

Reply:
got it, that's unfortunate.